src/pages/Dashboard/Profile.jsx

```diff
@@ -1,13 +1,252 @@
-import React from 'react';
+import React, { useState } from 'react';
+import styled from 'styled-components';
 import { AddressCard } from 'styled-icons/fa-regular/AddressCard';
 import { DashboardPage } from '../../components';
+import {
+    Button,
+    Card,
+    Checkbox,
+    Heading,
+    Input,
+    Select,
+} from '@cheapreats/react-ui';
+
+const testUser = {
+    status: 'PENDING',
+    email: 'jacky.chan@fake.utoronto.ca',
+    firstname: 'Jacky',
+    lastname: 'Chan',
+    gender: 'MALE',
+    school: 'UTSC',
+    bio: 'actor/fighter',
+    photo: 'https://i.ibb.co/HF6q3BF/doggo.gif',
+    links: [
+        {
+            name: 'github',
+            href: 'https://github.com/cheapreats/react-ui-library',
+        },
+        {
+            name: 'linkedin',
+            href: 'https://github.com/cheapreats/react-ui-library',
+        },
+        {
+            name: 'website',
+            href: 'https://github.com/cheapreats/react-ui-library',
+        },
+    ],
+};
 
 export const Profile = () => {
-    return <DashboardPage heading="Profile"></DashboardPage>;
+    const [user, setUser] = useState({
+        ...testUser,
+        currentPassword: '',
+        newPassword: '',
+        confirmPassword: '',
+    });
+    const onChange = e => setUser({ ...user, [e.target.name]: e.target.value });
+    const onChangeLinks = e => {
+        const i = user.links.findIndex(l => l.name === e.target.name);
+        const links = [...user.links];
+        links[i] = { ...links[i], href: e.target.value };
+        setUser({ ...user, links });
+    };
+
+    const onSubmitGeneralInformation = e => {
+        e.preventDefault();
+        const updatedUser = {
+            firstname: user.firstname,
+            lastname: user.lastname,
+            gender: user.gender,
+            school: user.school,
+            bio: user.bio,
+            photo: user.photo,
+            links: user.links,
+        };
+        console.log(updatedUser);
+    };
+    const onSubmitEmail = e => {
```

Comment:
Try onRegister and onSignin to make these more clear

Reply:
These functions are used to change the email and password, not register/sign in.
I changed their names to onSubmitEmailChange and onSubmitPasswordChange if that makes it more clear